DEBUGGING.md

```diff
@@ -0,0 +1,71 @@
+# Debugging the OpenTofu code
+
+There are various ways to debug the OpenTofu code. There is no ultimate "right" answer, this document intends to collect some of those ways. The order of debugging techniques is completely random.
+
+If you would like to contribute to this debugging guide, please [create a GitHub issue](https://github.com/opentofu/opentofu/issues/new/choose) and propose the enhancement. After that, you can create a pull request and reference this issue in your PR.
+
+For further information on contributing to the code, please refer to the [CONTRIBUTING.md](./CONTRIBUTING.md) file.
+
+<!-- TOC -->
```

Comment:
@Yantrio I'd like to keep this header as `TOC` instead of `MarkdownTOC` if there is no objection. (=